app/models/host/base.rb

```diff
@@ -399,7 +399,7 @@ def set_interface(attributes, name, iface)
       iface.attrs = attributes
 
       logger.debug "Saving #{name} NIC for host #{self.name}"
-      result = iface.save
+      result = iface.save if iface.new_record? || iface.changed?
```

Comment:
Thanks, done. I noticed that the whole block should have been within the conditional as it was also checking for errors after save. It's all been moved.